aws-cpp-sdk-core/source/client/CoreErrors.cpp

```diff
@@ -164,6 +166,10 @@ AWSError<CoreErrors> CoreErrorsMapper::GetErrorForName(const char* errorName)
   {
     return AWSError<CoreErrors>(CoreErrors::REQUEST_TIME_TOO_SKEWED, true);
   }
+  else if (errorHash == REQUEST_TIMEOUT_HASH)
+  {
+    return AWSError<CoreErrors>(CoreErrors::REQUEST_TIMEOUT, true);
```

Comment:
Timeout errors are not retry-able. Change the last argument to `false`

Reply:
I've seen a fair number of S3 calls time out (usually over unauthenticated HTTP) and then work fine straight afterwards. I currently have to overlay timeout retries one layer up from the SDK but it'd be nice for this to at least be optional. Or if not, can you elaborate on why they aren't retriable?

Reply:
HTTP 408 Request Timeout response status code means that the server would like to shut down this unused connection. It's basically an error on the client side. All 4xx errors are not retry-able by the SDK.